riposte-metrics-codahale-signalfx/src/main/java/com/nike/riposte/metrics/codahale/contrib/SignalFxReporterFactory.java

```diff
@@ -0,0 +1,94 @@
+package com.nike.riposte.metrics.codahale.contrib;
+
+import com.nike.riposte.metrics.codahale.ReporterFactory;
+
+import com.codahale.metrics.MetricRegistry;
+import com.signalfx.codahale.reporter.SignalFxReporter;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+/**
+ * A {@link ReporterFactory} for {@link SignalFxReporter}.
+ *
+ * @author Nic Munroe
+ */
+@SuppressWarnings("WeakerAccess")
+public class SignalFxReporterFactory implements ReporterFactory {
+
+    protected SignalFxReporter signalFxReporter;
+    protected final String signalFxApiKey;
+    protected final Function<SignalFxReporter.Builder, SignalFxReporter.Builder> reporterConfigurator;
+
+    /**
+     * Creates a new instance with the given SignalFx API key (auth token) and no custom reporter configuration.
+     * If you want to adjust the settings on the {@link SignalFxReporter.Builder} before the SignalFx reporter is
+     * created, then you should use the {@link SignalFxReporterFactory#SignalFxReporterFactory(String, Function)}
+     * constructor instead.
+     *
+     * @param signalFxApiKey The SignalFx API key (auth token) to use when reporting data to SignalFx.
+     */
+    public SignalFxReporterFactory(String signalFxApiKey) {
+        this(signalFxApiKey, null);
+    }
+
+    /**
+     * Creates a new instance with the given SignalFX API key (auth token) and given reporter configurator function.
+     * The SignalFx {@link SignalFxReporter.Builder} will be passed through the given function before building the
+     * reporter, allowing you to customize anything you want about the reporter. For example:
+     *
+     * <pre>
+     *      new SignalFxReporterFactory("mySignalFxApiKey", (reporterConfig) ->
+     *          reporterConfig.addDimension("foo", "bar")
+     *                        .addUniqueDimension("host_foo", "host_bar")
+     *                        .setDetailsToAdd(SignalFxReporter.MetricDetails.ALL)
+     *      );
+     * </pre>
+     *
+     * <p>If you pass in null for the reporterConfigurator argument then {@link Function#identity()} will be used for
+     * a fully default SignalFx reporter.
+     *
+     * <p>You can adjust the {@link #getInterval()} and {@link #getTimeUnit()} methods if you want the SignalFx reporter
+     * to report on a different schedule than the 10 second default.
```

Comment:
Is it possible to adjust these without a code change? Should these be made configurable?